### DISCUSSION TITLE
Integrate IdeaVim-sneak plugin 👟

### DISCUSSION THREAD

**Discussion by Mishkun** [JetBrains/ideavim#818] [Ideas]
I'm stopping maintenance of https://github.com/Mishkun/ideavim-sneak. But this plugin was quite popular with 23k+ dowloads so It makes sense to integrate it to the IdeaVim itself

**Comment by AlexPl292** [comment]
Hi everyone, the ideavim-sneak plugin is now a [part](https://github.com/JetBrains/ideavim/blob/master/doc/IdeaVim%20Plugins.md#sneak) of IdeaVim 🎉 .
We've made the release 2.8.4 with this plugin included, now it should work out of the box. The existing IntelliJ plugin `ideavim-sneak` can be removed.

**Reply by hboon** [reply]
Thanks! Is it possible to configure it like this, as we can with `vim-sneak` ?

```
map f <Plug>Sneak_s
map F <Plug>Sneak_S
omap z <Plug>Sneak_s
omap Z <Plug>Sneak_S
```

**Reply by citizenmatt** [reply]
It looks like the code is expecting braces. E.g.: `<Plug>(sneak-s)`

**Reply by AlexPl292** [reply]
Hi! I see that the mappings for the sneak plugin are broken. I fixed the mappings and supported both vim-sneak (`Sneak_s`) and the previous plugin mappings `(sneak-s)`. This should be available with the new release of the plugin.

**Reply by hboon** [reply]
Sorry for taking so long to get to trying this again (I had held off upgrading IdeaVim for a long time now to reduce chances of it breaking)

This works great. Thanks!

```
map f <Plug>Sneak_s
map F <Plug>Sneak_S
```

2 more things:

A. I have this in my Neovim config, any chance of getting it to work with IdeaVim?

```
omap <silent> f :call sneak#wrap(v:operator, 2, 0, 1, 1)<CR>
```

Given the text:

```
[cursor]here I am typing some text
```

Without that `omap`, typing `dfam`, the output is:

```
am typing some text
```

It works more like the `t` than `f` movement operator. It's a little inconsistent behavior with Sneak itself which is more like a 2-character `f`.

So with that `omap` in Neovim, typing `dfam`, the output is:

```
m typing some text
```

The "a" is cleared too.

B. `f<a><CR>` doesn't seem to work (In Neovim, the <CR> forces it to look for the 1-character "a")

Please let me know if it's better to write these elsewhere? Thanks.

**Reply by AlexPl292** [reply]
Hi there!
A: This is not possible now, but theoretically it's possible to implement this for IdeaVim. But I'm afraid we won't have the capacity to implement this in the near future.
B: This is also not implemented, but this seems easier to implement. Can't say any ETA, though.

**Reply by hboon** [reply]
No worries. B sounds like it'd be more generally useful and "standard" for others, so that's good if we could have that sometime in the future when you come around to it. Thanks in advance.

**Reply by limonoct** [reply]
@AlexPl292  how to enable highlighting?

**Reply by liango2** [reply]
🚀 Request: Implement Feature B
Body:
Hi there,
Could you please help implement Feature B? This functionality is critical for our workflow, and we're eager to see it integrated. We truly appreciate your time and effort in prioritizing this.
Thank you for your ongoing support!

**Comment by bstadlbauer** [comment]
This is great, thanks so much!

Would there be a possibility to include "label mode", similar to the original https://github.com/justinmk/vim-sneak plugin?

**Reply by AlexPl292** [reply]
Hi there. Sorry, no nearest plans for this feature. But I'd be happy to accept a PR if anyone will have time to implement that.

**Comment by asayake-b5** [comment]
Thanks for including and taking over the maintenance!

Would it be possible to include this [commit which was never tagged to a release](https://github.com/AlexPl292/ideavim-sneak/commit/0a230f62ba8b5f7d67082fc188b3ee51694350e8)? It fixes a conflict between the sneak and surround plugins.

**Reply by AlexPl292** [reply]
Yep, no problems. I'll move it, but could you please point to the problem that will be fixed by that? TBH, I don't see the case when the conflict happens...

**Reply by dimsuz** [reply]
@AlexPl292 I believe it's from [this PR](https://github.com/Mishkun/ideavim-sneak/pull/26) which resolves [this issue](https://github.com/Mishkun/ideavim-sneak/issues/25)

**Reply by AlexPl292** [reply]
This change is actually already merged into the ideavim-sneak plugin.

**Comment by gyaaniguy** [comment]
Is it me, or is highlighting not implemented yet ?
.My ideavimrc

```
set belloff=all
set visualbell
nnoremap <Home> ^
Plug 'justinmk/vim-sneak'
Plug 'tpope/vim-surround'
set hlsearch
set ignorecase
set relativenumber
set number

set incsearch
map Q gq
Plug 'machakann/vim-highlightedyank'
Plug 'tpope/vim-commentary'

```

**Reply by limonoct** [reply]
how to enable highlighting here?

**Comment by patrickwerz** [comment]
Thre is a delay when pressing `s` until the plugin is responsive. Also, there is a delay when jumping with `,` and `;`. Can this be solved?

**Comment by liango2** [comment]
1. Can Pinyin be supported like IntelliJ AceJump's setting: (map Unicode to ASCII)？
2. Can support f {char} or F {char} across multiple lines？
